lib/coffee.js

```diff
@@ -211,11 +219,22 @@ class Coffee extends EventEmitter {
     cmd.once('close', this.emit.bind(this, 'close'));
 
     if (this.stdin.length) {
-      this.stdin.forEach(function(buf) {
-        debug('input stdin `%s`', show(buf));
-        cmd.stdin.write(buf);
-      });
-      cmd.stdin.end();
+      if (this._isWaitForPrompt) {
+        // wait for message then write to stdin
+        cmd.on('message', msg => {
+          if (msg.type !== 'prompt' || this.stdin.length === 0) return;
+          const buf = this.stdin.shift();
+          debug('prompt stdin `%s`', show(buf));
+          cmd.stdin.write(buf + '\n');
```

Comment:
Do we really need to force a line break (`\n`)  here? Maybe the user want to control it by himself, for example, write [ANSI escape code](https://en.wikipedia.org/wiki/ANSI_escape_code) to emit special keypress events, etc.

Reply:
see usage at https://github.com/node-modules/common-boilerplate/pull/1/files#diff-0fd0e07cf6d02bf7cf00f18cebb8e6eaR50

I think more the time, developer had to add `\n`.

anyway, this is the point I'm unable to decide.

Reply:
Yeah, a trade-off, hard decision to make.

Reply:
had changed